src/main/java/org/datavyu/plugins/nativeosx/AVFoundationMediaPlayer.java

```diff
@@ -0,0 +1,94 @@
+package org.datavyu.plugins.nativeosx;
+
+
+import org.datavyu.plugins.MediaError;
+import org.datavyu.plugins.MediaException;
+import org.datavyu.plugins.PlayerStateEvent;
+import org.datavyu.plugins.PlayerStateListener;
+
+import java.awt.*;
+import java.io.File;
+import java.net.URI;
+
+public class AVFoundationMediaPlayer extends NativeOSXMediaPlayer {
+
+  private Container container;
+  private final Object readyLock = new Object();
+
+  public AVFoundationMediaPlayer(URI mediaPath, Container container) {
+    super(mediaPath);
+    this.addMediaPlayerStateListener(new AVFoundationMediaPlayer.PlayerStateListenerImpl());
+    this.container = container;
+  }
+
+  @Override
+  public void init() {
+    initNative();
+
+    // Test if he file exists
+    File mediaFile = new File(mediaPath);
+    if (!mediaFile.exists()){
+      throwMediaErrorException(MediaError.ERROR_MEDIA_INVALID.code(), null);
+    }
+
+    container.addNotify();
+    container.add(nativePlayerCanvas, BorderLayout.CENTER);
+
+    incPlayerCount();
+    sendPlayerStateEvent(eventPlayerReady, 0);
+
+    synchronized (readyLock) {
+      try {
+        readyLock.wait();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
+  }
+
+  @Override
+  protected void playerSeek(double streamTime, int flags) throws MediaException {
+    switch (flags){
+      case NORMAL_SEEK_FLAG:
+        EventQueue.invokeLater(() -> nativePlayerCanvas.setTime((long) streamTime * 1000, id));
+        break;
+      case PRECISE_SEEK_FLAG:
+        EventQueue.invokeLater(() -> nativePlayerCanvas.setTimePrecise((long) streamTime * 1000, id));
+        break;
+      case MODERATE_SEEK_FLAG:
+        EventQueue.invokeLater(() -> nativePlayerCanvas.setTimeModerate((long) streamTime * 1000, id));
+        break;
+    }
+  }
+
+  private class PlayerStateListenerImpl implements PlayerStateListener {
+
+    @Override
+    public void onReady(PlayerStateEvent evt) {
+      synchronized (readyLock){
```

Comment:
What happens if the onReady event fires during the playback.  E.g. the player goes through the following sequence of states: PLAYING, STALLED, PLAYING?  I'm not sure if the playerSetStartTime(0) should happen in that case.  Also, the may not be need to set as well -- it might change the size unintentionally.